cpp/oneapi/dal/algo/decision_forest/common.hpp

```diff
@@ -459,6 +459,72 @@ class descriptor : public detail::descriptor_base<Task> {
     }
 };
 
+/// Struct containing base description of node in descision tree
+struct node_descriptor {
+    std::int64_t level = 0; /// Number of connections between the node and the root
+    double impurity =
+        0.; /// Measure of the homogeneity of the response variable at the node (i.e., the value of the criterion)
+    std::int64_t node_sample_count = 0; /// Number of samples at the node
+};
+
+/// Struct containing description of split node in descision tree
+struct split_node_descriptor : public node_descriptor {
+    std::int64_t feature_index = 0; /// Feature used for splitting the node
+    double feature_value = 0.; /// Threshold value at the node
+};
+
+template <typename Task>
+struct leaf_node_descriptor;
+
+/// Struct containing description of leaf node in classification descision tree
+template <>
+struct leaf_node_descriptor<task::classification> : public node_descriptor {
+    std::int64_t label = 0; /// Label to be predicted when reaching the leaf
+    const double* prob = nullptr; /// Probabilities estimation for the leaf
+};
+
+/// Struct containing description of leaf node in regression descision tree
+template <>
+struct leaf_node_descriptor<task::regression> : public node_descriptor {
+    double label = 0.; /// Label to be predicted when reaching the leaf
+};
+
+template <typename T>
+struct is_leaf_node_descriptor {
+    static constexpr bool value = std::is_same_v<T, leaf_node_descriptor<task::classification>> ||
+                                  std::is_same_v<T, leaf_node_descriptor<task::regression>>;
+};
+
+template <typename T>
+struct is_split_node_descriptor {
+    static constexpr bool value = std::is_same_v<T, split_node_descriptor>;
+};
+
+template <typename T>
+inline constexpr bool is_leaf_node_descriptor_v = is_leaf_node_descriptor<T>::value;
+
+template <typename T>
+inline constexpr bool is_split_node_descriptor_v = is_split_node_descriptor<T>::value;
+
+} // namespace v1
+
+namespace detail {
+namespace v1 {
+
+template <typename Task>
+class tree_node_visitor_iface;
+
+template <typename Task, typename Op>
+std::shared_ptr<tree_node_visitor_iface<Task>> get_tree_node_visitor(Op&& op);
+} // namespace v1
+
+using v1::tree_node_visitor_iface;
+using v1::get_tree_node_visitor;
```

Comment:
This must be part of  `tree_node_visitor.hpp`

Reply:
fixed